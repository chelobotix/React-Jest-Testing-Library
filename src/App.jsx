import {useState} from 'react'

function App() {
    const [counter, setCounter] = useState(0);
    return (
        <>
            <h1>Works!</h1>
            <h2>React Jest Test</h2>
            <h3 data-testid="h3_Id">Test by data-testid</h3>
            <p>{counter}</p>
            <button type="button" onClick={() => setCounter(counter + 1)}>Increment</button>
        </>
    )
}

export default App
