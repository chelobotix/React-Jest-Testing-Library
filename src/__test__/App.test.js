import { render, screen, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'; //snapshot

import App from '../App'

describe('App Testing', () => {
  test('Check if render', () => {
    render(<App />)
    const h1 = screen.getByRole('heading', { level: 1 })
    expect(h1).toBeInTheDocument();
  })

  test('Check text content', () => {
    render(<App />)
    const h1 = screen.getByText(/Works!/i)
    expect(h1).toHaveTextContent('Works!');
  })

  test('Heading type + text', () => {
    render(<App />)
    const h2 = screen.getByRole('heading', { name: 'React Jest Test' })
    expect(h2).toHaveTextContent('React Jest Test');
  })

  test('By Id', () => {
    render(<App />)
    const h3 = screen.getByTestId("h3_Id")
    expect(h3).toHaveTextContent('Test by data-testid');
  })

  test('Get all by role', () => {
    render(<App />)
    const headingArray = screen.getAllByRole('heading');
    expect(headingArray.length).toBe(3);
  })

  test('Click button', () => {
    render(<App />)
    const button = screen.getByText('Increment')
    const countElement = screen.getByText('0');
    fireEvent.click(button)
    expect(countElement.textContent).toBe('1');
  })

  test('Tree', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot()
  })
})
