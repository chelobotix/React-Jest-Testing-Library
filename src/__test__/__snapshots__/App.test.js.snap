// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading type Tree 1`] = `
[
  <h1>
    Works!
  </h1>,
  <h2>
    React Jest Test
  </h2>,
  <h3
    data-testid="h3_Id"
  >
    Test by data-testid
  </h3>,
  <p>
    0
  </p>,
  <button
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>,
]
`;
